stacks/api/function/runtime/test/node/entrypoint.spec.ts

```diff
@@ -296,11 +297,205 @@ describe("Entrypoint", () => {
       const exitCode = await runtime
         .execute({
           eventId: event.id,
-          cwd: event.target.cwd
+          cwd: event.target.cwd,
+          stdout: "inherit"
         })
         .catch(e => e);
 
       expect(exitCode).toBe(4);
     });
   });
+
+  describe("Firehose", () => {
+    let firehoseQueue: FirehoseQueue;
+
+    beforeEach(() => {
+      queue.drain();
+      firehoseQueue = new FirehoseQueue();
+      queue.addQueue(firehoseQueue);
+      queue.listen();
+    });
+
+    it("pop from the queue", async () => {
+      await initializeFn(`export default function({socket, pool}, message) {
+        console.log(arguments);
+        if ( 
+          socket.remoteAddress == "[::1]" && 
+          socket.id == "1" &&
+          pool.size == 21 &&
+          message.name == "test" && 
+          message.data == "test"
+        ) {
+          process.exit(4);
```

Comment:
@thesayyn What's the meaning of '4'?

Reply:
Um. Actually, it's just an exit code but let me give you the background. 

When a process exits with a "non-zero (a number that is not 0)"  it means the process has failed. 
So you may ask why I didn't exit with code 0 instead of 4?

It's because by default the process will quit with "0 exit code" when the program executes successfully so I wanted to check explicitly that the condition meets and the  process exits with code 4.